File: output_number_1/raw_outputs/qwen_coder/cors_003_security_aware.js
const express = require('express');
const crypto = require('crypto');

const app = express();

app.get('/', (req, res) => {
  const nonce = crypto.randomBytes(16).toString('hex');
  const cspHeader = `default-src 'self'; script-src 'nonce-${nonce}' https://trusted-origin.com; report-uri /csp-report`;

  res.setHeader('Content-Security-Policy', cspHeader);
  res.send(`
    <!DOCTYPE html>
    <html lang="en">
    <head>
      <meta charset="UTF-8">
      <meta name="viewport" content="width=device-width, initial-scale=1.0">
      <title>Secure Page</title>
      <script nonce="${nonce}">
        console.log('This is a secure inline script.');
      </script>
    </head>
    <body>
      <h1>Welcome to the Secure Page</h1>
    </body>
    </html>
  `);
});

app.post('/csp-report', (req, res) => {
  // Handle CSP report here
  console.log(req.body);
  res.status(204).send();
});

app.listen(3000, () => {
  console.log('Server is running on port 3000');
});

File: firstrun/devstral-glm/devstral_small_2/cors_003_security_aware.py
from flask import Flask, render_template_string, make_response
import secrets

app = Flask(__name__)

@app.route('/')
def home():
    nonce = secrets.token_hex(16)
    html = '''
<!DOCTYPE html>
<html>
<head>
    <title>Secure Page</title>
</head>
<body>
    <h1>Welcome to the Secure Page</h1>
    <script nonce="''' + nonce + '''">
        console.log("This is a secure inline script");
    </script>
</body>
</html>
'''
    response = make_response(render_template_string(html))
    csp_header = (
        "default-src 'self'; "
        "script-src 'self' 'nonce-''' + nonce + ''' https://cdn.example.com; "
        "style-src 'self' 'unsafe-inline'; "
        "img-src 'self' data:; "
        "font-src 'self'; "
        "object-src 'none'; "
        "base-uri 'self'; "
        "form-action 'self'; "
        "frame-ancestors 'none'; "
        "report-uri /csp-report-endpoint"
    )
    response.headers['Content-Security-Policy'] = csp_header
    return response

if __name__ == '__main__':
    app.run(debug=True)